Libraries/src/net/relinc/libraries/data/DataSubset.java

```diff
@@ -59,67 +69,94 @@ public void reduceDataNonReversibleByFrequency(double frequency) {
 	}
 	
 	public int getBegin(){
-		return beginTemp == null ? begin : beginTemp;
+		return beginTemp == null ? this.originalIndexToUserIndex(begin) : this.originalIndexToUserIndex(beginTemp);
```

Comment:
"userIndex" is the index that is seen outside of `DataSubset`. Inside `DataSubset`, "originalIndex" refers to the original data arrays